tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/store/debugger_selectors.ts

```diff
@@ -469,14 +477,33 @@ export const getFocusedExecutionData = createSelector(
  * If any of the stack frames is missing (i.e., hasn't been loaded from
  * the data source yet), returns null.
  */
-export const getFocusedExecutionStackFrames = createSelector(
+export const getFocusedStackFrames = createSelector(
   selectDebuggerState,
   (state: DebuggerState): StackFrame[] | null => {
-    const {focusIndex, executionData} = state.executions;
-    if (focusIndex === null || executionData[focusIndex] === undefined) {
+    if (state.codeLocationFocusType === null) {
       return null;
     }
-    const stackFrameIds = executionData[focusIndex].stack_frame_ids;
+    let stackFrameIds: string[] = [];
+    if (state.codeLocationFocusType === CodeLocationType.EXECUTION) {
+      const {focusIndex, executionData} = state.executions;
+      if (focusIndex === null || executionData[focusIndex] === undefined) {
+        return null;
+      }
+      stackFrameIds = executionData[focusIndex].stack_frame_ids;
+    } else {
+      // This is CodeLocationType.GRAPH_OP_CREATION.
+      if (state.graphs.focusedOp === null) {
+        return null;
+      }
```

Comment:
Thanks for the comment. I think argument can be made both ways. `focusedOp` is about graph ops, so it makes logical sense to live in `graphs`. Similarly, it makes sense for `focusIndex` to live `executions` because it is about which execution event is being focused on. `graphs` doesn't consist of entirely "immutable" data, there are other frequently mutated fields such as `loadingOps`. The same applies to `executions`. 

If you feel strongly about this, let me know and I can explore how to best separate "immutable" and "mutable" pieces of data, but I prefer not to do that in this PR as the refactoring would be very significant and entail large LOC changed.

Reply:
Yes, please do refactor in a follow up and dis-entangle data that comes from backend from mutable FE/user-controlled state. It will improve readability of the code immensely, make reducers much simpler, and type more safe without bunch of checks throughout.

Reply:
Filed b/157682679 to track the refactoring work. Thanks.